### DISCUSSION TITLE
Creating additional options for buttons and selectMenus

### DISCUSSION THREAD

**Discussion by aidanhstewart** [discordjs/discord.js#7957] [Ideas]
On buttons and selectMenus, it would be nice if there were an option to make it so only certain roles, permissions or users can see the button or option in a selectMenu

**Comment by imranbarbhuiya** [comment]
You should suggest in https://discord.com/discord-api-docs also there is already a feature request opened.
https://github.com/discord/discord-api-docs/discussions/3344 for this feature
